File: integrations/wasm/zig/tests/generated_contract.zig
const std = @import("std");
const angular = @import("angular-ts");
const player = @import("player-contract");

test "generated fields carry paths and Zig value types" {
    try std.testing.expectEqualStrings("position.x", player.positionX.path);
    try std.testing.expectEqual(f64, @TypeOf(player.positionX).Value);
    try std.testing.expectEqual(u32, @TypeOf(player.health).Value);
    try std.testing.expectEqualStrings("frame", player.frame.path);
    try std.testing.expect(player.frame.is_optional);

    const scope = angular.Scope.fromHandle(1);
    _ = scope;
}

File: integrations/wasm/rust/crates/angular-ts/tests/generated_contract.rs
use angular_ts::{BinaryField, Field};

mod player {
    include!(concat!(
        env!("CARGO_MANIFEST_DIR"),
        "/../../../contracts/generated/player_contract.rs"
    ));
}

fn accepts_field<T>(_: Field<T>) {}

#[test]
fn generated_contract_fields_preserve_rust_value_types() {
    accepts_field::<f64>(player::POSITION_X);
    accepts_field::<f64>(player::POSITION_Y);
    accepts_field::<u32>(player::HEALTH);
    accepts_field::<String>(player::NAME);

    let frame: BinaryField = player::FRAME;
    assert_eq!(frame.path(), "frame");
    assert!(frame.is_optional());
}
